Modules/_ctypes/_ctypes.c

```diff
@@ -4635,7 +4635,7 @@ Array_subscript(PyObject *myself, PyObject *item)
             wchar_t *dest;
 
             if (slicelen <= 0)
-                return PyUnicode_New(0, 0);
+                return Py_NewRef(&_Py_STR(empty));
```

Comment:
Out of curiosity, what is the benefit of this change except inline calling?
AFAIK, `PyUnicode_New(0, 0);` will finally call `unicode_get_empty` which is the same replacement of `_Py_STR(empty)`

https://github.com/python/cpython/blob/f33e2c87a83917b5139d97fd8ef7cba7223ebef5/Objects/unicodeobject.c#L259-L271

Reply:
AFAIK, @ericsnowcurrently try to reduce private API from the stdlib module as the best practice for 3rd party users who try to write C extension module by referencing stdlib implementation..
So I don't think this is a good replacement if there are no significant improvements.

Please let me know if I know wrong Eric :)

Reply:
That makes sense. I had first seen that ceval.c could change, so I made similar changes elsewhere, but if the other code is better left alone, I don't mind.

I updated this PR to only have the ceval.c change, where the code can actually get smaller and have fewer branches.